src/Build/BackEnd/Components/ProjectCache/ProjectCacheBase.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Execution;
+
+namespace Microsoft.Build.Experimental.ProjectCache
+{
+    /// <summary>
+    ///     Only one plugin instance can exist for a given BuildManager BeginBuild / EndBuild session.
+    /// </summary>
+    public abstract class ProjectCacheBase
```

Comment:
This should probably be named something like `ProjectCachePluginBase` or something more precise, this name seems too generic to me.

Reply:
Yeah, `ProjectCachePluginBase` sounds better.